plugins/Previous/src/Previous.ts

```diff
@@ -0,0 +1,45 @@
+import { Plugin, PluginCallbacks, State, self, PluginCallbacksOnSetArgument } from '@hookstate/core';
+
+const PluginID = Symbol('Previous');
+
+type PreviousInstanceType<S = any> = {
+  (): PluginCallbacks & {
+    getPrevious(): S
+  }
+  previous: S | undefined
+}
+
+const PreviousInstance: PreviousInstanceType = () => ({
+  onSet(data: PluginCallbacksOnSetArgument) {
+    PreviousInstance.previous = data.previous
```

Comment:
data.previous is a previous value at path not previous value at root. you plugin would work only with primitive global/local states, but would not handle nested states.

Reply:
I can implement with lodash.clonedeep like Initial plugin, it will solve the problem?